server.js

```diff
@@ -240,23 +240,29 @@ app.prepare().then(async () => {
       publicOnly: true,
     });
 
-    const exploreSlates = await Data.getSlatesByIds("id", [
-      //NOTE(tara): slates in localhost for testing
-      "857ad84d-7eff-4861-a988-65c84b62fc23",
-      "81fa0b39-0e96-4c7f-8587-38468bb67cb3",
-      "c4e8dad7-4ba0-4f25-a92a-c73ef5522d29",
-      "df05cb1f-2ecf-4872-b111-c4b8493d08f8",
-      "435035e6-dee4-4bbf-9521-64c219a527e7",
-      "ac907aa3-2fb2-46fd-8eba-ec8ceb87b5eb",
-
-      //NOTE(tara): slates in prod
-      // "d2861ac4-fc41-4c07-8f21-d0bf06be364c",
-      // "9c2c458c-d92a-4e81-a4b6-bf6ab4607470",
-      // "7f461144-0647-43d7-8294-788b37ae5979",
-      // "f72c2594-b8ac-41f6-91e0-b2da6788ae23",
-      // "a0d6e2f2-564d-47ed-bf56-13c42634703d",
-      // "0ba92c73-92e7-4b00-900e-afae4856c9ea",
-    ]);
+    let exploreSlates = [];
+
+    if (Environment.IS_PRODUCTION) {
+      exploreSlates = await Data.getSlatesByIds([
```

Comment:
Did you test this?

`Data.getSlatesByIds` can't change unless you change the function declaration too.

Reply:
yep I restart the server and it works. i think the current method should just pass in one array. not sure why the previous one works as well tho.

Reply:
https://github.com/filecoin-project/slate/blob/main/node_common/data/methods/get-slates-by-ids.js because you're not actually using the function the way its written?

Reply:
what do you mean? 
I think `Data.getSlatesByIds` currently should only take an array as "id". so I think there is no need to modify the function?

Reply:
it's just curious to me that taking two arguments `Data.getSlatesByIds("id", [1,2,3,4,5])` works the same as `Data.getSlatesByIds([1,2,3,4,5])`

Reply:
```
export default async ({ id }) => {
  return await runQuery({
    label: "GET_SLATES_BY_IDS",
    queryFn: async (DB) => {
      const query = await DB.select("*").from("slates").whereIn("id", [
        //NOTE(tara): slates in localhost for testing
        "857ad84d-7eff-4861-a988-65c84b62fc23",
        "81fa0b39-0e96-4c7f-8587-38468bb67cb3",
        "c4e8dad7-4ba0-4f25-a92a-c73ef5522d29",
        "df05cb1f-2ecf-4872-b111-c4b8493d08f8",
        "435035e6-dee4-4bbf-9521-64c219a527e7",
        "ac907aa3-2fb2-46fd-8eba-ec8ceb87b5eb",

        //NOTE(tara): slates in prod
        // "d2861ac4-fc41-4c07-8f21-d0bf06be364c",
        // "9c2c458c-d92a-4e81-a4b6-bf6ab4607470",
        // "7f461144-0647-43d7-8294-788b37ae5979",
        // "f72c2594-b8ac-41f6-91e0-b2da6788ae23",
        // "a0d6e2f2-564d-47ed-bf56-13c42634703d",
        // "0ba92c73-92e7-4b00-900e-afae4856c9ea",
      ]);

      if (!query || query.error) {
        return [];
      }

      return JSON.parse(JSON.stringify(query));
    },
    errorFn: async (e) => {
      return {
        error: true,
        decorator: "GET_SLATES_BY_IDS",
      };
    },
  });
};
```

* This function takes an object as an argument. The argument is spread so it looks for an `id`.
* Currently the way it is used is `Data.getSlatesByIds({ id: 'some-value' })`
* The way you have it written, it doesn't matter what you pass in it, you can delete `({ id })` and replace it with `()` and it would still do the same thing.
* Try it out for yourself, and if the function takes no arguments, then you can't pass things into it.

Reply:
Yes the reason why you could literally do anything, you could pass it `Data.getSlatesByIds('beach', 'forest', 'mountain', 'jim-is-lame')` is because the way you have the function written right now it does not care what is passed in.

Reply:
Here is a function that does the exact same thing

```
export default async () => {
  return await runQuery({
    label: "GET_SLATES_BY_IDS",
    queryFn: async (DB) => {
      const query = await DB.select("*").from("slates").whereIn("id", [
        //NOTE(tara): slates in localhost for testing
        "857ad84d-7eff-4861-a988-65c84b62fc23",
        "81fa0b39-0e96-4c7f-8587-38468bb67cb3",
        "c4e8dad7-4ba0-4f25-a92a-c73ef5522d29",
        "df05cb1f-2ecf-4872-b111-c4b8493d08f8",
        "435035e6-dee4-4bbf-9521-64c219a527e7",
        "ac907aa3-2fb2-46fd-8eba-ec8ceb87b5eb",

        //NOTE(tara): slates in prod
        // "d2861ac4-fc41-4c07-8f21-d0bf06be364c",
        // "9c2c458c-d92a-4e81-a4b6-bf6ab4607470",
        // "7f461144-0647-43d7-8294-788b37ae5979",
        // "f72c2594-b8ac-41f6-91e0-b2da6788ae23",
        // "a0d6e2f2-564d-47ed-bf56-13c42634703d",
        // "0ba92c73-92e7-4b00-900e-afae4856c9ea",
      ]);

      if (!query || query.error) {
        return [];
      }

      return JSON.parse(JSON.stringify(query));
    },
    errorFn: async (e) => {
      return {
        error: true,
        decorator: "GET_SLATES_BY_IDS",
      };
    },
  });
};
```

Reply:
Maybe writing what you wrote like this will help make it more clear

```
export default async () => {
  return await runQuery({
    label: "GET_SLATES_BY_IDS",
    queryFn: async (DB) => {
      const query = await DB.select("*").from("slates").whereIn("id", [
        "857ad84d-7eff-4861-a988-65c84b62fc23",
        "81fa0b39-0e96-4c7f-8587-38468bb67cb3",
        "c4e8dad7-4ba0-4f25-a92a-c73ef5522d29",
        "df05cb1f-2ecf-4872-b111-c4b8493d08f8",
        "435035e6-dee4-4bbf-9521-64c219a527e7",
        "ac907aa3-2fb2-46fd-8eba-ec8ceb87b5eb",
      ]);

      if (!query || query.error) {
        return [];
      }

      return JSON.parse(JSON.stringify(query));
    },
    errorFn: async (e) => {
      return {
        error: true,
        decorator: "GET_SLATES_BY_IDS",
      };
    },
  });
};
```

Reply:
If that isn't clear, this is what you have done

```
const addTwoNumbers = (a, b) => {
  return 2 + 2;
}
```

const result = addTwoNumbers(2, 2);
console.log(result); // 4

still works right? what happens when you change it to `addTwoNumbers(3, 3);`

Reply:
I see what you mean, basically the whole thing is passed in as id

Reply:
You have total control of the preconditions (another word: arguments) you pass into a function.

* How it is declared
* At the call site, such as `await Data.doSomethingFun()` what you pass through.

Your code won't throw an error even if you did something like: `await Data.doSomethingFun(32, true, null, ['dogs', 'cats'])`

Reply:
Another thing:

```
const addTwoNumbers = (a, b) => {
  return a + b;
}

const addTwoNumbers = ({ a, b }) => {
  return a + b;
}

const addTwoNumbers = (options) => {
  return options.a + options.b;
}
```

Think about how these functions are used:

```
addTwoNumbers(2, 2);
addTwoNumbers({ a: 2, b: 2 });
addTwoNumbers({ a: 2, b: 2 });
```

Does this make sense?

Reply:
ok so I will need to make it so that it ONLY takes in one array?

Reply:
When we were discussing earlier the usage of this function, now that you understand what I am saying, look at the other files and think of the way to keep the usage of this "type of function" the same. This function is unique because it expects a `set` instead of a `single argument`.

Reply:
Regardless, `{ id }` is wrong for a function named `Data.getSlatesByIds` right? because `id` is singular it suggests theres only one `id`, when you would use this function for a set of them.

Reply:
hopefully this works :]